### PR TITLE
Switch to self hosted runner

### PR DESCRIPTION
**Pull request by printminion-co**
* Resolves: # 

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by muhali-ionos** [comment]
Check now, as the public repos need permissions to run the self-hosted runner at the organization level

File: .github/workflows/hidrive-next-build.yml [hunk 1]
```diff
@@ -60,6 +60,23 @@ jobs:
         with:
           node-version-file: "package.json"
 
+      - name: Install Dependencies
+        run: sudo apt-get update && sudo apt-get install -y make zip unzip
+
+      - name: Verify dependencies versions
```
Comment (thlehmann-ionos) [inline .github/workflows/hidrive-next-build.yml:66]: Please reword `Verify` to `Print`

File: .github/workflows/hidrive-next-build.yml [hunk 2]
```diff
@@ -60,6 +60,23 @@ jobs:
...
+
+      - name: Setup PHP with PECL extension
+        uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231 #v2.31.1
+        with:
+          tools: composer:v2
+          extensions: gd, zip, curl, xml, xmlrpc, mbstring, sqlite, xdebug, pgsql, intl, imagick, gmp, apcu, bcmath, redis, soap, imap, opcache
+        env:
+          runner: self-hosted
+
+      - name: Verify PHP install
```
Comment (thlehmann-ionos) [inline .github/workflows/hidrive-next-build.yml:77]: Please reword `Verify` to `Print`

**Review by thlehmann-ionos — APPROVED** [review]
Looks good to me.

**Pull request merged**